### PR TITLE
feat(widget-library): Add custom widget tab

### PR DESCRIPTION
**Pull request by shruthilayaj**
Reuse the custom widget modal in the widget
library tab. Clicking on "Custom" button allows
user to add a widget through the regular flow.
The library and custom widget states are preserved
as the user switches between the two "tabs"

Design change: Replaced the "two tab" design
with a single "Custom" tab and "Back to Library"
from the latest feedback round.

Update: Fixes the icons blowing up in Firefox.

More tests to follow.

Screen Shot 2021-11-15 at 11 56 37 AM
d
Screen Shot 2021-11-15 at 11 56 43 AM

### CODE REVIEW

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 1]
```diff
@@ -552,24 +583,49 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
...
-            </Button>
-          </ButtonBar>
+            <ButtonBar gap={1}>
+              {fromLibrary ? (
+                <Button
+                  data-test-id="add-widget"
+                  type="button"
+                  onClick={() => {
+                    if (dashboard && onAddLibraryWidget) {
+                      openDashboardWidgetLibraryModal({
```
Comment (shruthilayaj) [inline static/app/components/modals/addDashboardWidgetModal.tsx:600]: Do I need to call `closeModal()` before opening a new one here?

**Comment by Jesse-Box** [comment]
@shruthilayaj I think I found the culprit to your icon issue...

specifying a value of `"small"` doesn't point to any pixel size, but if you remove the `size `prop entirely it should fix the problem.

**Review by edwardgou-sentry — APPROVED** [review]
LGTM! Just one question about the `Back to Library` button

File: static/app/components/modals/addDashboardWidgetModal.tsx (hunk 1 above)
Comment (shruthilayaj) [reply static/app/components/modals/addDashboardWidgetModal.tsx:608]: Yeah I inspected the HTML that gets rendered, there isn't any weird nesting as far as I can tell, but just wanted to double check that I wasn't missing anything and we didn't keep mounting these modals on the DOM or something!

File: static/app/components/modals/addDashboardWidgetModal.tsx [hunk 2]
```diff
@@ -560,24 +591,49 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
...
-              {isUpdatingWidget ? t('Update Widget') : t('Add Widget')}
-            </Button>
-          </ButtonBar>
+            <ButtonBar gap={1}>
+              {fromLibrary ? (
+                <Button
+                  data-test-id="back-to-library"
+                  type="button"
+                  onClick={() => {
+                    if (dashboard && onAddLibraryWidget) {
```
[the remark this answers is not in the document]
Comment (shruthilayaj) [reply static/app/components/modals/addDashboardWidgetModal.tsx]: Updated in 15fdaa649f8628203652bda152a410782dac6059

**Pull request merged**